ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * A representation of an Ambry user. A user is an entity (an application or an individual user) who uses ambry
+ * as service. This class contains general information of a user, which can be used for user-based operations
+ * such as authentication, get a {@link Container} under this account, and access control. The account name is
+ * provided by an Ambry user as an external reference. Account id is an internal identifier of the user, and is
+ * one-to-one mapped to the account name. Account name and id are generated through user registration process.
+ * Account id is part of a blob’s properties, and cannot be modified once the blob is created. Account metadata
+ * is made in JSON, which is generic to contain additional information of the metadata.
+ *
+ * Version 1 of account metadata in JSON is in the format below.
+ *  {
+ *    "accountId": 101,
+ *    "accountName": "MyAccount",
+ *    "containers": [
+ *      {
+ *        "containerName": "MyPrivateContainer",
+ *        "description": "This is my private container",
+ *        "isPublic": "true",
+ *        "containerId": 0,
+ *        "version": 1,
+ *        "status": "active"
+ *      },
+ *      {
+ *        "containerName": "MyPublicContainer",
+ *        "description": "This is my public container",
+ *        "acl": "false",
+ *        "containerId": 1,
+ *        "version": 1,
+ *        "status": "active"
+ *      }
+ *    ],
+ *    "version": 1,
+ *    "status": "active"
+ *  }
+ */
+public class Account {
+  // static variables
+  public static final String ACCOUNT_METADATA_VERSION_KEY = "version";
+  public static final String ACCOUNT_ID_KEY = "accountId";
+  public static final String ACCOUNT_NAME_KEY = "accountName";
+  public static final String ACCOUNT_STATUS_KEY = "status";
+  public static final String CONTAINERS_KEY = "containers";
+  public static final String ACCOUNT_STATUS_ACTIVE = "active";
```

Comment:
For both `Account` and `Container` class, their status are made as String and some values are predefined as constant. It should be ok for now coz the status is not in a closed definition so far, and there is no complex logic to process the states. Will be moving to a more rigorous definition if needed.